lms/djangoapps/instructor/enrollment.py

```diff
@@ -286,36 +291,51 @@ def reset_student_attempts(course_id, student, module_state_key, requesting_user
             text_type(module_state_key),
         )
 
-    module_to_reset = StudentModule.objects.get(
-        student_id=student.id,
-        course_id=course_id,
-        module_state_key=module_state_key
-    )
-
-    if delete_module:
-        module_to_reset.delete()
-        create_new_event_transaction_id()
-        set_event_transaction_type(grades_events.STATE_DELETED_EVENT_TYPE)
-        tracker.emit(
-            six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
-            {
-                'user_id': six.text_type(student.id),
-                'course_id': six.text_type(course_id),
-                'problem_id': six.text_type(module_state_key),
-                'instructor_id': six.text_type(requesting_user.id),
-                'event_transaction_id': six.text_type(get_event_transaction_id()),
-                'event_transaction_type': six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
-            }
-        )
-        if not submission_cleared:
-            _fire_score_changed_for_block(
-                course_id,
-                student,
-                block,
-                module_state_key,
+    def _reset_or_delete_module(studentmodule):
+        if delete_module:
+            studentmodule.delete()
+            create_new_event_transaction_id()
+            set_event_transaction_type(grades_events.STATE_DELETED_EVENT_TYPE)
+            tracker.emit(
+                six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
+                {
+                    'user_id': six.text_type(student.id),
+                    'course_id': six.text_type(course_id),
+                    'problem_id': six.text_type(module_state_key),
+                    'instructor_id': six.text_type(requesting_user.id),
+                    'event_transaction_id': six.text_type(get_event_transaction_id()),
+                    'event_transaction_type': six.text_type(grades_events.STATE_DELETED_EVENT_TYPE),
+                }
             )
+            if not submission_cleared:
+                _fire_score_changed_for_block(
+                    course_id,
+                    student,
+                    block,
+                    module_state_key,
+                )
+        else:
+            _reset_module_attempts(studentmodule)
+
+    if teams_enabled:
+        from lms.djangoapps.teams.api import get_team_for_user_course_topic
+        team = get_team_for_user_course_topic(student, str(course_id), selected_teamset_id)
+        if team:
```

Comment:
what if teams are enabled but the user isn't on a team? should probably fall through to erasing their own module.